cheroot/ssl/pyopenssl.py

```diff
@@ -328,13 +335,38 @@ def wrap(self, sock):
         # closing so we can't reliably access protocol/client cert for the env
         return sock, self._environ.copy()
 
+    def _password_callback(
+        self,
+        password_max_length,
+        _verify_twice,
+        password,
+        /,
+    ):
+        """Pass a passphrase to password protected private key."""
+        b_password = b''
+        if isinstance(password, str):
+            b_password = password.encode('utf-8')
+        elif isinstance(password, bytes):
+            b_password = password
+
+        if len(b_password) > password_max_length:
+            _warn(
+                f'User-provided password is {len(b_password)} bytes long and will '
```

Comment:
I'd rather not have length computed twice. Also, complex expressions reduce readability. Could you move this to a var?